ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/results/tracing/diff/StateTraceGenerator.java

```diff
@@ -85,129 +84,133 @@ private Stream<StateDiffTrace> generate(
       return Stream.empty();
     }
 
-    // Get the world state: transactionUpdater is post-transaction, previousUpdater is
-    // pre-transaction
     final WorldUpdater transactionUpdater = getTransactionUpdater(traceFrames);
-    final WorldUpdater previousUpdater = transactionUpdater.parentUpdater().get();
+    final WorldUpdater parentUpdater = transactionUpdater.parentUpdater().get();
 
     final StateDiffTrace stateDiffResult = new StateDiffTrace();
 
-    // Compare modified accounts (existing or new accounts touched by the tx)
+    // In pre-state mode, we capture the full pre-state of all accounts touched during execution.
+    // Only the parent updater is relevant here. Post-transaction state does not matter.
+    // To identify which accounts were read, we rely on the transaction trace’s access list (BAL).
+    if (isPreState) {
+      processPreStateMode(parentUpdater, stateDiffResult, transactionTrace);
```

Comment:
Think it would be clearer that preState is only using the transaction trace for the touched accounts if only the touched accounts were passed in as a argument

Reply:
minor nitpick really. But what meant was changing to something like this `processPreStateMode(parentUpdater, stateDiffResult, transactionTrace.getTouchedAccounts());`